iml-agent/src/action_plugins/snapshot.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{agent_error::ImlAgentError, env};
+use iml_wire_types::LdevEntry;
+use tokio::{
+    fs::File,
+    io::AsyncWriteExt,
+};
+
+async fn write_to_file(content: String) -> Result<(), ImlAgentError> {
+    let ldev_path = env::get_var("LDEV_CONF_PATH");
+
+    let mut file = File::create(ldev_path).await?;
+    file.write_all(content.as_bytes()).await?;
+
+    Ok(())
+}
+
+async fn create_ldev_conf_internal<F>(entries: Vec<LdevEntry>, write_to_file: impl Fn(String) -> F) -> Result<(), ImlAgentError> 
+where F: futures::Future<Output=Result<(), ImlAgentError>>
+{
+    let content = entries
+        .iter()
+        .map(|x| x.to_string())
+        .collect::<Vec<String>>()
+        .join("\n");
+    
+    write_to_file(content).await?;
+
+    Ok(())
+}
+
+pub async fn create_ldev_conf(entries: Vec<LdevEntry>) -> Result<(), ImlAgentError> {
+    create_ldev_conf_internal(entries, write_to_file).await?;
+    
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[tokio::test]
+    async fn test_create_ldev_conf() -> Result<(), ImlAgentError> {
+        // oss2 oss1 zfsmo-OST0013 zfs:ost19/ost19
```

Comment:
This could be an inline test (without a snapshot file) :)

Reply:
It could, I like using snapshots though :)

Reply:
Snapshots can be inlined as well (not that I'm advocating for one over the other).